tests/providers/microsoft/azure/sensors/test_wasb.py

```diff
@@ -128,9 +128,17 @@ def create_context(self, task, dag=None):
         deferrable=True,
     )
 
-    def test_wasb_blob_sensor_async(self):
+    @mock.patch("airflow.providers.microsoft.azure.sensors.wasb.WasbHook")
+    @mock.patch("airflow.providers.microsoft.azure.sensors.wasb.WasbBlobSensor.defer")
+    def test_wasb_blob_sensor_finish_before_deferred(self, mock_defer, mock_hook):
+        mock_hook.return_value.check_for_blob.return_value = True
+        self.SENSOR.execute(mock.MagicMock())
+        assert not mock_defer.called
```

Comment:
should we use `mock_defer.assert_not_called()` instead?

Reply:
yes. I think we should

Reply:
Done in [4d9fefc](https://github.com/apache/airflow/pull/31009/commits/4d9fefc43fb92a99735b0fb65c23728b2498fc5e)

Reply:
it doesnt work as `mock_defer.assert_not_called()` results in None